der/src/length/indefinite.rs

```diff
@@ -81,6 +84,7 @@ pub(crate) fn read_eoc<'a>(reader: &mut impl Reader<'a>) -> crate::Result<()> {
 pub(crate) fn read_constructed_vec<'r, R: Reader<'r>>(
     reader: &mut R,
     header: Header,
```

Comment:
Since `inner_tag` has taken over the role of `header.tag`, perhaps this should just be the length:

```suggestion
    length: Length,
```